test_scripts/API/VehicleData/HandsOffSteering/common.lua

```diff
@@ -0,0 +1,369 @@
+  local handsOffSteeringResponseData = {
+    dataType = "VEHICLEDATA_HANDSOFFSTEERING",
+    resultCode = "SUCCESS"
+  }
+  m.getMobileSession(pAppId):StartService(7)
+  :Do(function()
+    local params = m.cloneTable(m.getParams(pAppId))
+    params.hashID = m.getHashId(pAppId)
+    local corId = m.getMobileSession(pAppId):SendRPC("RegisterAppInterface", params)
+    m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppRegistered", {
+      application = { appName = m.getParams(pAppId).appName }
+    })
+    m.getMobileSession(pAppId):ExpectResponse(corId, { success = true, resultCode = "SUCCESS" })
+    :Do(function()
+      m.getMobileSession(pAppId):ExpectNotification("OnPermissionsChange")
+    end)
+  end)
+  if true == isHMISubscription then
+    m.getHMIConnection():ExpectRequest( "VehicleInfo.SubscribeVehicleData", { handsOffSteering = true })
+    :Do(function(_, data)
+      m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS",
+        { handsOffSteering = handsOffSteeringResponseData })
+    end)
+  else
+    m.getHMIConnection():ExpectRequest( "VehicleInfo.SubscribeVehicleData", { handsOffSteering = true }):Times(0)
+  end
+end
+
+--[[ @onVehicleData: Processing OnVehicleData notification
+--! @parameters:
+--! pHandsOffSteering: Vehicle parameter value
+--! pExpTimes:  expected number of notifications
+--! pAppId: application number (1, 2, etc.)
+--! @return: none
+--]]
+function m.onVehicleData(pHandsOffSteering, pExpTimes, pAppId)
+  if not pExpTimes then pExpTimes = 1 end
+  if not pAppId then pAppId = 1 end
+  m.getHMIConnection():SendNotification("VehicleInfo.OnVehicleData", { handsOffSteering = pHandsOffSteering })
+  m.getMobileSession(pAppId):ExpectNotification("OnVehicleData", { handsOffSteering = pHandsOffSteering })
+  :Times(pExpTimes)
+end
+
+--[[ @onVehicleDataForTwoApps: Processing OnVehicleData notification for two apps
+--! @parameters:
+--! pHandsOffSteering: Vehicle parameter value
+--! pAppId_1, pAppId_2: application number (1, 2, etc.)
+--! @return: none
+--]]
+function m.onVehicleDataForTwoApps(pHandsOffSteering, pAppId_1, pAppId_2)
+  m.getHMIConnection():SendNotification("VehicleInfo.OnVehicleData", { handsOffSteering = pHandsOffSteering })
+  m.getMobileSession(pAppId_1):ExpectNotification("OnVehicleData", { handsOffSteering = pHandsOffSteering })
+  m.getMobileSession(pAppId_2):ExpectNotification("OnVehicleData", { handsOffSteering = pHandsOffSteering })
+end
+
+--[[ @unexpectedDisconnect: closing connection
+--! @parameters: none
+--! @return: none
+--]]
+function m.unexpectedDisconnect()
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = true })
+  :Times(actions.mobile.getAppsCount())
+  actions.mobile.disconnect()
+  utils.wait(1000)
+end
+
+--[[ @postcondition: Stop SDL and restore sdl_preloaded_pt.json file
+ --! @parameters: none
+ --! @return: none
+ --]]
+function m.postcondition()
+  m.restorePreloadedPT()
```

Comment:
According to comment from Dmytro restore is performed automatically in backup function is called before, so please remove custom postcondition function and use it from actions